File: src/debit.cpp
#include "include/debit.h"


big_int debit::get_cash_size() const {
  return cash_size_;
}

void debit::set_bill_id(const big_int &id) {
  bill_id_ = id;
}

big_int debit::get_cashback_potential_size() const {
  return cashback_potential_size_;
}

std::string debit::get_bill_kind() const {
  return BILL_KIND_DEBIT;
}

big_int debit::get_bill_id() const {
  return bill_id_;
}

big_int debit::get_person_id() const {
  return person_id_;
}

void debit::set_cash_size(const big_int &cash_size) {
  cash_size_ = cash_size;
}

void debit::set_cashback_potential_size(const big_int &cashback_potential_size) {
  cashback_potential_size_ = cashback_potential_size;
}

void debit::set_person_id(const big_int &person_id) {
  person_id_ = person_id;
}

void debit::update() {
  cashback_potential_size_ = cash_size_ / 100 * 5;
}

void debit::assign_id() {
  std::ifstream file("max_bill_id.txt");
  std::string s;
  file >> s;
  bill_id = big_int(s);
  if (bill_id == 0) {
    bill_id++;
    bill_id_ = 0;
  } else bill_id_ = ++bill_id;

  file.close();
  std::ofstream file1("max_bill_id.txt");
  file1 << bill_id.toString();
  file1.close();
}

File: MIPT_bank_app1/MIPT_bank_app/app/src/main/java/com/example/mipt_bank_app/Debit.java
package com.example.mipt_bank_app;

import java.security.PublicKey;

public class Debit extends IBill {

    @Override
    public String getBillKind() {
        return bill_kind_;
    }

    @Override
    public int getCashSize() {
        return cash_size_;
    }

    public int getCashbackPotentialSize() {
        return cashback_potential_size_;
    }

    @Override
    public int getBillId() {
        return bill_id_;
    }

    @Override
    public int getPersonId() {
        return person_id_;
    }

    @Override
    public void setCashSize(int cash_size) {
        cash_size_ = cash_size;
    }

    @Override
    public void setBillId(int id) {
        bill_id_ = id;
    }

    public void setCashbackPotentialSize(int cashback_potential_size) {
        cashback_potential_size_ = cashback_potential_size;
    }

    @Override
    public void setPersonId(int person_id) {
        person_id_ = person_id;
    }

    public void update() {
        cashback_potential_size_ = cash_size_ / 100 * 5;
    }

    @Override
    public void assignId() {

    }

    private String bill_kind_ = SupConstants.BILL_KIND_DEBIT;
    private int cashback_potential_size_;
}
